rust/core-lib/src/tabs.rs

```diff
@@ -370,8 +370,10 @@ impl CoreState {
         let view_id = self.next_view_id();
         let buffer_id = self.next_buffer_id();
 
+        let mut is_existing_file = false;
+
         let rope = match path.as_ref() {
-            Some(p) => self.file_manager.open(p, buffer_id)?,
+            Some(p) => {is_existing_file = true; self.file_manager.open(p, buffer_id)?},
```

Comment:
oh also: `rustfmt` (which i'll turn on soon) is going to make us put this on separate lines.